crates/storage/provider/src/traits/chain.rs

```diff
@@ -0,0 +1,140 @@
+///! Canonical chain state notification trait and types.
+use crate::PostState;
+use auto_impl::auto_impl;
+use core::fmt;
+use reth_primitives::{BlockNumHash, BlockNumber, Receipt, SealedBlockWithSenders, TxHash};
+use std::{collections::BTreeMap, fmt::Formatter, sync::Arc};
+use tokio::sync::broadcast::{Receiver, Sender};
+
+/// Trait that holds blocks and post state of execution those blocks.
+#[auto_impl(&, Arc)]
+pub trait SubChain: Send + Sync {
+    /// Get chain post state.
+    fn state(&self) -> &PostState;
```

Comment:
is this the state after the `last` block in `blocks`?

Reply:
Yes, it is the state after executing all blocks.